apps/src/blockly/addons/plusMinusBlocks/advancedProcedures.js

```diff
@@ -690,3 +610,196 @@ const procedureVars = function () {
 };
 
 GoogleBlockly.Extensions.register('procedure_vars', procedureVars);
+
+/**
+ * Validates text entered into the argument name field.
+ * @param {string} newValue The new text entered into the field.
+ * @returns {?string} The field's new value.
+ * @this {GoogleBlockly.FieldTextInput}
+ */
+function argumentReporterValidator(newValue) {
+  const sourceArgBlock = this.getSourceBlock();
+  const defBlock = sourceArgBlock.getRootBlock();
+  if (
+    !Blockly.cdoUtils.isFunctionBlock(defBlock) ||
+    !sourceArgBlock.isShadow()
+  ) {
+    return;
+  }
+
+  const argName =
+    sourceArgBlock.outputConnection?.targetConnection?.getParentInput()?.name;
+  const workspace = sourceArgBlock.workspace;
+  const argData = defBlock.argData_;
+  const argDatum = defBlock.argData_.find(element => element.argId === argName);
+  const currId = argDatum.model.getId();
+
+  // Replace all whitespace with normal spaces, then trim.
+  newValue = newValue.replace(/[\s\xa0]+/g, ' ').trim();
+  const caselessName = newValue.toLowerCase();
+
+  /**
+   * Returns true if the given argDatum is associated with this field, or has
+   * a different caseless name than the argDatum associated with this field.
+   * @param {{model: GoogleBlockly.VariableModel, argId:string}} argDatum The
+   *     argDatum we want to make sure does not conflict with the argDatum
+   *     associated with this field.
+   * @returns {boolean} True if the given datum does not conflict with the
+   *     datum associated with this field.
+   * @this {GoogleBlockly.FieldTextInput}
+   */
+  const hasDifName = argDatum => {
```

Comment:
Could we have a more specific name for this? Maybe `isValidArgDatum` since having a different name is only one thing we check?

Reply:
That works! The previous name just came from the Blockly source code so I just hadn't considered improving the name.